### PR TITLE
project files restructuring

### PR DESCRIPTION
**Pull request by emmanueltct**
the following are changes made to my  project
1. change the folder structure of my file
2. creating middleware for each entity (blogs, comments, and Querries)
3. add dist folder to gitignore
4. implement validation for Cloudinary image uploader
5. customizing status code return message for my endpoints 
6. protecting my endpoints for authorized or authenticated access and making the public access to others

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,4 +1,11 @@
 PORT=5000
 dbURL="mongodb+srv://emmanuelmunezero:AndelaCohort29@cluster0.prj6gh4.mongodb.net/?retryWrites=true&w=majority"
```
Comment (Eli250) [inline .env:2]: You shouldn't be pushing the environment variables. Add this file in .gitignore file.

File: src/controllers/blogsComment.ts [hunk 2]
```diff
@@ -1,46 +1,47 @@
 import {Request,Response} from "express"
-import Blog from "../models/blog"
 import blogComment from '../models/blogComment'
 
 
 const createComment=async(req:Request,res:Response)=>{
     /* console.log(req.body.user)*/
     try{
+        let user=req.user
+        console.log(req.user)
```
Comment (Eli250) [inline src/controllers/blogsComment.ts]: Remove this unnecessary log.

File: src/controllers/blogsComment.ts [hunk 3]
```diff
@@ -1,46 +1,47 @@
...
+            const newComment=new blogComment({
+                blogId:req.params.id,
+                user:user,
+                message:req.body.message
+                })
+        
+                const comment= await newComment.save()
+                res.status(200).json({message:"Your comment to this blog is successful sent"})
+        }else{
+            res.status(400).json({error:"we are validating you before you place a comment and something went wrong"})
```
Comment (Eli250) [inline src/controllers/blogsComment.ts]: Your message doens't make sense. Also use proper http status code.

File: src/controllers/blogsComment.ts [hunk 4]
```diff
@@ -1,46 +1,47 @@
...
-
-        const comment= await newComment.save()
-        res.send(comment)
-   
-  
     }catch{
-        res.status(404)
-        res.send({error:"some thing went wrong with your comment"})
+        res.status(500).json({error:"some thing went wrong with your comment"})
     }
```
Comment (Eli250) [inline src/controllers/blogsComment.ts:26]: Always make sure that you return a response. use 'return' to mark the end of the process.

File: src/controllers/blogsComment.ts [hunk 5]
```diff
@@ -1,46 +1,47 @@
...
 const readComment=async(req:Request,res:Response)=>{
-    console.log(req.params.id)
     try {
         const post = await blogComment.find({blogId: req.params.id })
-        res.send(post)
+        res.status(200).json({data:post})
     } catch {
-        res.status(404)
-        res.send({ error: "Post doesn't exist!" })
+        res.status(400).json({ error: "comments doesn't exist!" })
```
Comment (Eli250) [inline src/controllers/blogsComment.ts]: Wrong http status code.

File: src/controllers/blogsComment.ts [hunk 6]
```diff
@@ -1,46 +1,47 @@
...
-    console.log( req.params.comment_id)
     try {
         const comment=await blogComment.findOne({ _id: req.params.comment_id })
-        res.send(comment)
+        res.status(200).json({data:comment})
     } catch {
-        res.status(404)
-        res.send({ error: "Post doesn't exist!" })
+        res.status(400).json({ error: "Comment doesn't exist!" })
     }
```
Comment (Eli250) [inline src/controllers/blogsComment.ts:49]: You are not using the right http status. It has to be 404 in case it is not found.

**Review by Eli250 — CHANGES_REQUESTED** [review]

**Review by emmanueltct — COMMENTED** [review]
I made changes to my project according to feedback review and the following are changes in my code
1. add .env file to gitignore
2. customize my status code to match with endpoint response
3. add a return key word to each end point response

**Pull request merged**